### PR TITLE
Allow user-controlled FIFOs

### PR DESCRIPTION
**Pull request by theopolis**
This removes the check for user controlled FIFOs and instead allows reads of pseudo files up until a configuration-defined max limit. There may be ways to improve the read performance, this could use some integration tests and benchmarking.

### CODE REVIEW

File: osquery/filesystem/filesystem.cpp [hunk 1]
```diff
@@ -145,8 +143,25 @@ Status readFile(const fs::path& path,
...
+  if (file_size == 0) {
+    off_t total_bytes = 0;
+    ssize_t part_bytes = 0;
+    do {
+      auto part = std::string(4096, '\0');
+      part_bytes = read(handle.fd, &part[0], 4096);
+      if (part_bytes > 0) {
+        total_bytes += part_bytes;
+        if (total_bytes >= read_max) {
+          VLOG(1) << "FIFO file too large";
```
Comment (theopolis) [inline osquery/filesystem/filesystem.cpp]: This also applies to pseudo files types.

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) failed :-1:

**Comment by osqueryer** [comment]
Build (5d91076148758d306d08b2cbacc3a25ff8a26c3f) is good! :+1:

**Pull request merged**